framework/src/auxkernels/MaterialStdVectorAux.C

```diff
@@ -34,5 +37,8 @@ MaterialStdVectorAux::~MaterialStdVectorAux()
 Real
 MaterialStdVectorAux::getRealValue()
 {
+  mooseAssert(_special_qp < _q_point.size(), "MaterialStdVectorAux.  special_qp specified as " << _special_qp << " but there are only " << _q_point.size() << " quadpoints in the element");
```

Comment:
I just tried the NaN, and my exodiff doesn't work.  Exodiff runs and produces:

Element variables:
materials/types.testNaN: 	ERROR: NaN found for variable stdvec0_qp0 in block 0, file 1

So you can see i outputted NaN to the exodus file, but exodiff thinks that's an error.  What do you think i should do, @dschwen ?